pkg/actors/actors.go

```diff
@@ -687,71 +687,62 @@ func (a *actorsRuntime) updateReminderTrack(actorKey, name string, repetition in
+		nextInvokeTime, expirationTime time.Time
+		dueTime, period, ttl           time.Duration
+		repeats, repetitionsLeft       int
+	)
 
-	dueTime, err := time.ParseDuration(reminder.DueTime)
+	registeredTime, err := time.Parse(time.RFC3339, reminder.RegisteredTime)
 	if err != nil {
-		return nextInvokeTime, errors.Wrap(err, "error parsing reminder due time")
+		return errors.Wrap(err, "error parsing reminder registered time")
 	}
 
-	key := constructCompositeKey(reminder.ActorType, reminder.ActorID)
-	track, err := a.getReminderTrack(key, reminder.Name)
-	if err != nil {
-		return nextInvokeTime, errors.Wrap(err, "error getting reminder track")
+	if len(reminder.DueTime) != 0 {
+		if dueTime, err = time.ParseDuration(reminder.DueTime); err != nil {
+			return errors.Wrap(err, "error parsing reminder due time")
+		}
 	}
 
-	var lastFiredTime time.Time
-	if track != nil && track.LastFiredTime != "" {
-		lastFiredTime, err = time.Parse(time.RFC3339, track.LastFiredTime)
-		if err != nil {
-			return nextInvokeTime, errors.Wrap(err, "error parsing reminder last fired time")
+	repeats = -1
+	if len(reminder.Period) > 0 {
+		if period, repeats, err = parseDuration(reminder.Period); err != nil {
+			return errors.Wrap(err, "error parsing reminder period")
 		}
 	}
 
-	// the first execution reminder task
-	if lastFiredTime.IsZero() {
-		nextInvokeTime = registeredTime.Add(dueTime)
-	} else {
-		period, _, err := parseDuration(reminder.Period)
-		if err != nil {
-			return nextInvokeTime, errors.Wrap(err, "error parsing reminder period")
+	if len(reminder.TTL) > 0 {
+		if ttl, _, err = parseDuration(reminder.TTL); err != nil {
+			return errors.Wrap(err, "error parsing reminder TTL")
 		}
-		nextInvokeTime = lastFiredTime.Add(period)
+		expirationTime = registeredTime.Add(ttl).Add(dueTime)
 	}
 
-	return nextInvokeTime, nil
-}
-
-func (a *actorsRuntime) startReminder(reminder *Reminder, stopChannel chan bool) error {
-	actorKey := constructCompositeKey(reminder.ActorType, reminder.ActorID)
-	reminderKey := constructCompositeKey(actorKey, reminder.Name)
-	nextInvokeTime, err := a.getUpcomingReminderInvokeTime(reminder)
+	track, err := a.getReminderTrack(actorKey, reminder.Name)
 	if err != nil {
-		return err
+		return errors.Wrap(err, "error getting reminder track")
 	}
 
-	var period time.Duration
-	repetitionsLeft := -1
-	if reminder.Period != "" {
-		period, repetitionsLeft, err = parseDuration(reminder.Period)
+	if track != nil && len(track.LastFiredTime) != 0 {
+		lastFiredTime, err := time.Parse(time.RFC3339, track.LastFiredTime)
 		if err != nil {
-			log.Errorf("error parsing reminder period %s: %s", reminder.Period, err)
-			return err
+			return errors.Wrap(err, "error parsing reminder last fired time")
 		}
-		log.Debugf("repetitions allowed are %d", repetitionsLeft)
+		repetitionsLeft = track.RepetitionLeft
+		nextInvokeTime = lastFiredTime.Add(period)
+	} else {
+		repetitionsLeft = repeats
+		nextInvokeTime = registeredTime.Add(dueTime)
```

Comment:
Per my other component, by supporting ISO8601 duration start time we can have an alternative format to specify the first invoke time.

The logic should be something like:
```go
if (startTime != nil && startTime => registeredTime.Add(dueTime)) {
  nextInvokeTime = startTime
} else {
  nextInvokeTime = registeredTime.Add(dueTime)
}
```